prime-router/src/main/kotlin/azure/EmailSenderFunction.kt

```diff
@@ -0,0 +1,140 @@
+import com.sendgrid.Method
+import com.sendgrid.Request
+import com.sendgrid.Response
+import com.sendgrid.SendGrid
+import com.sendgrid.helpers.mail.Mail
+import com.sendgrid.helpers.mail.objects.Content
+import com.sendgrid.helpers.mail.objects.Email
+import com.sendgrid.helpers.mail.objects.Personalization
+import gov.cdc.prime.router.secrets.SecretHelper
+import java.io.IOException
+import java.util.logging.Logger
+
+const val NO_REPLY_EMAIL = "no-reply@cdc.gov"
+const val REGISTER_EMAIL = "reportstream@cdc.gov"
+const val REGISTER_SUBJECT_BASE = "TOS Agreement for "
+const val TEMPLATE_ID = "d-472779cf554f418a9209acb62d2a48da"
+
+class SenderTosRequest {
+    data class SenderTosFormJSON(
+        val title: String,
+        val firstName: String,
+        val lastName: String,
+        val email: String,
+        val territory: String,
+        val organizationName: String,
+        val operatesInMultipleStates: Boolean,
+        val agreedToTermsOfService: Boolean
+    )
+}
+
+class EmailSenderFunction {
+
+    @FunctionName("emailRegisteredOrganization")
+    @StorageAccount("AzureWebJobsStorage")
+    fun emailRegisteredOrganization(
+        @HttpTrigger(
+            name = "emailRegisteredOrganization",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-registered"
+        )
+        request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        val logger: Logger = context.logger
+        var ret = request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+
+        if (request.body !== null) {
+            logger.info(request.body)
+            ret.status(sendRegistrationConfirmation(request.body!!, logger))
+        }
+
+        return ret.build()
+    }
+
+    private fun sendRegistrationConfirmation(requestBody: String, logger: Logger): HttpStatus {
+        val gson = Gson()
+        val body = gson.fromJson<SenderTosRequest.SenderTosFormJSON>(
+            requestBody,
+            SenderTosRequest.SenderTosFormJSON::class.java
+        )
+        val title = body.title
+        val firstName = body.firstName
+        val lastName = body.lastName
+        val email = body.email
+        val territory = body.territory
+        val organizationName = body.organizationName
+        val operatesInMultipleStates = body.operatesInMultipleStates
+        val agreedToTermsOfService = body.agreedToTermsOfService
+
+        var status: HttpStatus = HttpStatus.NOT_FOUND
+        val content: Content = Content()
+        content.type = "plain/text"
+
+        val p: Personalization = Personalization()
+        p.addTo(Email(REGISTER_EMAIL))
+        p.addCc(Email(email))
+        p.addDynamicTemplateData(
+            "formData",
+            object {
+                var title = title
+                var firstName = firstName
+                var lastName = lastName
+                var email = email
```

Comment:
Is there any sanity check that it's a single, valid email? Most likely this is handled by SendGrid?

Reply:
SendGrid has an endpoint to validate emails: https://docs.sendgrid.com/api-reference/e-mail-address-validation/validate-an-email

What I'd honestly love to do is rework this file into something that can be used for much more than this single type of email. I think it'd actually clean this whole thing up quite a bit and make it easier to understand.